tests/test-regressions.lisp

```diff
@@ -5,3 +5,34 @@
   "https://github.com/sharplispers/log4cl/issues/1"
   (log4cl:start-hierarchy-watcher-thread)
   (finishes (log4cl-impl::save-hook)))
+
+(deftest recursive-lock.issue#8.1 ()
+  "https://github.com/sharplispers/log4cl/issues/8"
+  (let ((output (with-output-to-string (*debug-io*)
+                  (with-package-log-hierarchy
+                    (reset-logging-configuration)
+                    (with-simple-restart (abort "Abort")
+                      (handler-bind
+                          ((error (lambda (condition)
+                                    (log:error "caught error: ~A" condition)
+                                    (abort))))
+                        (log:warn "~@{~A ~A~}" 1)))))))
+    (is (search (format nil "WARN - 1 ~@
+                             ERROR - caught error: ")
+                output))))
+
```

Comment:
This test doesn't look right (with regard to `is` assertion), locally I see
```
" <WARN> [14:08:41] log4cl fileDssrBB (xxx)
<ERROR> [14:08:41] log4cl fileDssrBB (xxx)
* caught error: 
Missing argument
\"~@{~A ~A~}\"
        ^
"
```

and this won't match what is covered in `format`. Same goes for the test below. I think that we should simply assert, that the code finishes (without causing error). Do I miss something?

Reply:
ah, I see, it is specially configured for tests. either way, shouldn't we just test, that locking attempt doesn't cause the deadlock in these situations? that will remove ccl failure as well (which has nothing to do with logging).